sdl2-vga-terminal/src/VgaTerminal.cpp

```diff
@@ -165,97 +193,100 @@ void VgaTerminal::writeXY(const uint8_t x, const uint8_t y, const std::string &s
  */
 VgaTerminal::terminalChar_t VgaTerminal::at(const uint8_t x, const uint8_t y) const noexcept
 {
-    return (x >= _viewPortWidth || y >= _viewPortHeight)
-        ? defaultNullChar
-        : _pGrid[(static_cast<size_t>(y) + _viewPortY) * mode.tw + x + _viewPortX]
-        ;
-}
-
-void VgaTerminal::render(const bool force)
-{
-    // TODO:
-    // for performances this if should not be done
-    // and slice the for loop without the curX,Y pos
-    // like: (j =0; j< _curY), (j=_curY+1; j<mode.th) => the i loop as usual
-    //       j=curY, (i=0; <_curX),/(i=_curX+1; i<mode.tw)
-    //       j=curY, i=curX
-    //
-    //       at that point the loops is a general function accepting the 4 vaues for the range
-    //       special case only for the cursor, 
-    //       and can just ignore the rendered flag
-    //
-
-    if (!force && (SDL_GetWindowFlags(getWindow()) & SDL_WINDOW_HIDDEN) == SDL_WINDOW_HIDDEN) {
-        return;
+    if (x >= _viewPortWidth || y >= _viewPortHeight) {
+        return defaultNullChar;
     }
 
-    // force to render the cursor everytime
-    int icur = static_cast<int>(_curY) * mode.tw + _curX;
-    _pGrid[icur].rendered = false;
+    _terminalChar_t _tc = _pGrid[(static_cast<size_t>(y) + _viewPortY) * mode.tw + x + _viewPortX];
+    terminalChar_t tc;
+    tc.c = _tc.c;
+    tc.col = _tc.col;
+    tc.bgCol = _tc.bgCol;
+ 
+    return tc;
+}
 
-    for (int j = 0; j < mode.th; j++) {
-         int j2 = j * mode.tw;
-         int jch = j * mode.ch;
-        
+void VgaTerminal::_renderGridPartialY(const uint8_t y1, const uint8_t y2, const bool force)
+{
+    for (int j = y1; j < y2; j++) {
+        int j2 = j * mode.tw;
+        int jch = j * mode.ch;
         for (int i = 0; i < mode.tw; i++) {
             int i2 = j2 + i;
 
             if (!force && _pGrid[i2].rendered) {
                 continue;
             }
-        
+
             SDL_Point p = { i * mode.cw, jch };
-            uint8_t col, bgCol, c;
-            c = _pGrid[i2].c;
-            col = _pGrid[i2].col;
-            bgCol = _pGrid[i2].bgCol;
-
-            // cursor position
-            if ((_curY == j) && (_curX == i)
-                && (showCursor && _cursonOn)) {
-                
-                if (c != 0) {
-                    // invert col and bgCol
-                    col = _pGrid[i2].bgCol;
-                    bgCol = _pGrid[i2].col;
-                }
-                else {
-                    // only the cursor
-                    bgCol = 0;
-                    col = curDefaultCol;
-                    c = cur_shape;
-                }
-            }
-            else {
-                _pGrid[i2].rendered = true;
-            }
 
-            renderChar(p, col, bgCol, c);
+            _renderFontChar(p, _pGrid[i2]);
+        }
+    }
+}
+
+void VgaTerminal::_renderGridLinePartialX(const uint8_t x1, const uint8_t x2, const bool force)
+{
+    int j2 = _curY * mode.tw;
+    int jch = _curY * mode.ch;
```

Comment:
converting to parameters and then make reusable this function:

`j2` => `y`
`jch` => `ych`